spec/test/stored_queries/test_query.py

```diff
@@ -0,0 +1,436 @@
+
+warnings.filterwarnings("ignore")
+
+# Directories and files
+ROOT_DIR = os.getcwd()
+CURR_DIR = os.path.join(ROOT_DIR, "spec/test/stored_queries")
+CONFIG_FP = os.path.join(ROOT_DIR, "arango_live_server_config.json")
+TEST_DATA_DIR = os.path.join(CURR_DIR, "../data")
+TMP_OUT_DIR = os.path.join(ROOT_DIR, "tmp")
+SCINAMES_LATEST_FP = os.path.join(TMP_OUT_DIR, "ncbi_scinames_latest.json")
+SAMPLINGS_FP = os.path.join(TMP_OUT_DIR, "samplings.json")
+STORED_QUERY_FP = os.path.join(
+    ROOT_DIR, "spec/stored_queries/taxonomy/taxonomy_ncbi_species.yaml"
+)
+STORED_QUERY_NO_SORT_FP = os.path.join(
+    ROOT_DIR, "spec/stored_queries/taxonomy/taxonomy_ncbi_species_no_sort.yaml"
+)
+
+if not os.path.exists(TMP_OUT_DIR):
+    os.mkdir(TMP_OUT_DIR)
+
+try:
+    with open(CONFIG_FP) as fh:
+        CONFIG = json.load(fh)
+    if not CONFIG["host"] or not CONFIG["username"] or not CONFIG["password"]:
+        raise RuntimeError("Missing config fields")
+    CLIENT = ArangoClient(hosts=CONFIG["host"])
+    DB = CLIENT.db("ci", username=CONFIG["username"], password=CONFIG["password"])
+except Exception as e:
+    help = """
+Please set host URL, username, and password in arango_live_server_config.json, e.g.,
+{
+    "username": "doe_j",
+    "password": "cat-sat-hat",
+    "host": "http://10.58.1.211:8532"
+}
+Note: if you are on a local machine
+you may have to proxy into the live ArangoDB server first, e.g.,
+`ssh -L 8532:10.58.1.211:8532 j_doe@login1.berkeley.kbase.us`
+Then, the url would be `http://localhost:8532`
+"""
+    print(help)
+    raise (e)
+NCBI_TAXON = DB.collection("ncbi_taxon")
+
+# Load the queries
+QUERY = json_validation.load_json_yaml(STORED_QUERY_FP)["query"]
+QUERY_NO_SORT = json_validation.load_json_yaml(STORED_QUERY_NO_SORT_FP)["query"]
+
+LIMIT = 20
+NOW = time.time() * 1000
+
+# Load/cache the scinames
+if os.path.isfile(SCINAMES_LATEST_FP):
+    with open(SCINAMES_LATEST_FP) as fh:
+        SCINAMES_LATEST = json.load(fh)
+else:
+    print("Fetching latest NCBI scinames ...")
+    try:
+        taxa_all = list(NCBI_TAXON.all())
+    except ReadTimeout:
+        print("Sorry, there is a read timeout. Please try again on a different machine")
+        sys.exit()
+    SCINAMES_LATEST = [
+        taxa["scientific_name"]
+        for taxa in taxa_all
+        if (taxa["rank"] in ["species", "strain"] or taxa["strain"])
+        and taxa["created"] <= NOW
+        and NOW <= taxa["expired"]
+    ]
+    with open(SCINAMES_LATEST_FP, "w") as fh:
+        json.dump(SCINAMES_LATEST, fh)
+
+
+def use_sort(search_text):
+    return len(search_text) > 3
+
+
+def is_simple(search_text):
+    """
+    Somewhat arbitrary determination of whether a fulltext's search text is "simple"
+    relative to its search time
+    """
+    return len(search_text.split()) == 2 and all(
+        [tok.isalnum() and len(tok) >= 3 for tok in search_text.split()]
+    )
+
+
+def jprint(jo, dry=False):
+    txt = json.dumps(jo, indent=3)
```

Comment:
I also like having `sort_keys` turned on to make it easier to find whatever you're looking for